ccm_web/server/src/api/api.admin.handler.ts

```diff
@@ -75,10 +78,16 @@ export class AdminApiHandler {
     try {
       const endpoint = `accounts/${accountId}/courses`
       logger.debug(`Sending request to Canvas (get all pages) - Endpoint: ${endpoint}; Method: GET`)
-      const courses = await this.requestor.listItems<CanvasCourse>(endpoint, queryParams).toArray()
+      const courses = []
+      const pages = this.requestor.listPages<CanvasCourse[]>(endpoint, queryParams)
+      for await (const courseResponse of pages) {
+        courses.push(...courseResponse.body)
+        if (courses.length > this.maxSearchCourses) throw new TooManyResultsError()
```

Comment:
This is nice, regardless if they searched by course name or uniqname the will be thrown and infact for both instructor and admin view.

Reply:
How is this? https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/379/commits/db632ac5ebbad736a6fcf615d44e9a066f681a7a

I was uncertain how best to do this without duplicating code too much; this is what I arrived at. There may be something better still.

Reply:
> This is nice, ~regardless if they searched by course name or uniqname the error will be thrown and infact for both instructor and admin view.~
> 
> Looks like the logic only applied to search by course name. I got the facts wrongs.

@pushyamig, I think you are partially correct. We use a different approach for instructor version of this feature, but if the admin user searches for a uniqname, this check still applies. I don't know if any individual teacher has >400 courses in a term, but that may be worth doing some research on.

Reply:
>I don't know if any individual teacher has >400 courses in a term

I feel chances are very slim, if the logic is working then I don't think we need to do more research into this.